### PR TITLE
Implemented the Dirty_Measure_Path on Android & iOS

### PR DESCRIPTION
**Pull request by carldebilly**
# Feature
_details coming..._

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/SamplesApp/UITests.Shared/Utilities/RainbowMeasures.cs [hunk 1]
```diff
@@ -11,91 +11,99 @@
...
+		protected override Size MeasureOverride(Size availableSize)
+		{
+			Background = new SolidColorBrush(_colors[++_measureIndex % _colors.Count]);
 
-            if (Children.Count > 0 && Children[0] is { } child)
-            {
-                child.Measure(availableSize);
+			if (Children.Count > 0 && Children[0] is { } child)
+			{
+#if XAMARIN
```
Comment (jeromelaban) [inline src/SamplesApp/UITests.Shared/Utilities/RainbowMeasures.cs:76]:
```suggestion
#if HAS_UNO
```

File: src/SamplesApp/UITests.Shared/Utilities/RainbowMeasures.cs [hunk 2]
```diff
@@ -11,91 +11,99 @@
...
+			}
+
+			return default;
+		}
+
+		protected override Size ArrangeOverride(Size finalSize)
+		{
+			if (Children.Count > 0 && Children[0] is { } child)
+			{
+#if XAMARIN
```
Comment (jeromelaban) [inline src/SamplesApp/UITests.Shared/Utilities/RainbowMeasures.cs:98]:
```suggestion
#if HAS_UNO
```

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml/UIElementTests/UIElement_MeasurePerf.xaml.cs [hunk 3]
```diff
@@ -149,7 +149,8 @@ private async void GoTest2(object sender, RoutedEventArgs e)
 
 			RainbowMeasures mostInner;
 
-			if (dept < deptness.Value)
+			var deptnessValue = (int)deptness.Value;
```
Comment (jeromelaban) [inline src/SamplesApp/UITests.Shared/Windows_UI_Xaml/UIElementTests/UIElement_MeasurePerf.xaml.cs:152]:
```suggestion
			var deepnessValue = (int)deepness.Value;
```

File: src/Uno.UI/UI/Xaml/Controls/Grid/Grid.Properties.cs [hunk 4]
```diff
@@ -131,7 +131,7 @@ private void OnCornerRadiusPropertyChanged(CornerRadius oldValue, CornerRadius n
 		#endregion
 
 		#region Row Property
-		[GeneratedDependencyProperty(DefaultValue = 0, AttachedBackingFieldOwner = typeof(UIElement), Attached = true, ChangedCallbackName = nameof(OnGenericPropertyChanged))]
+		[GeneratedDependencyProperty(DefaultValue = 0, Options = FrameworkPropertyMetadataOptions.AffectsMeasure | FrameworkPropertyMetadataOptions.AffectsArrange, AttachedBackingFieldOwner = typeof(UIElement), Attached = true)]
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/Grid/Grid.Properties.cs:134]: Are we invoking InvalidateMeasure somewhere in the callback that we should be removing?

File: src/Uno.UI/UI/Xaml/Controls/ScrollContentPresenter/NativeScrollContentPresenter.Android.cs [hunk 5]
```diff
@@ -103,32 +103,49 @@ private void InitializeScrollbars()
...
+			((ILayouterElement)this).OnMeasureInternal(widthMeasureSpec, heightMeasureSpec);
+		}
 
-			IFrameworkElementHelper.OnMeasureOverride(this);
+		void ILayouterElement.SetMeasuredDimensionInternal(int width, int height)
+		{
+			SetMeasuredDimension(width, height);
 		}
 
+		//TODO generated code
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/ScrollContentPresenter/NativeScrollContentPresenter.Android.cs:123]: To remove ?

File: src/Uno.UI/UI/Xaml/Controls/Grid/Grid.Properties.cs (hunk 4 above)
Comment (carldebilly) [reply src/Uno.UI/UI/Xaml/Controls/Grid/Grid.Properties.cs:134]: You can scroll horizontally and see I did that ;-)

File: src/Uno.UI/UI/Xaml/IFrameworkElement.cs [hunk 6]
```diff
@@ -206,6 +206,18 @@ public static void InvalidateMeasure(this IFrameworkElement e)
 				case View view:
 #if XAMARIN_ANDROID
 					view.RequestLayout();
+					var parent = view.Parent;
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/IFrameworkElement.cs:209]: Would be interesting to explain why this is needed.

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml/UIElementTests/UIElement_MeasurePerf.xaml [hunk 7]
```diff
@@ -13,14 +13,13 @@
 			<RowDefinition Height="*" />
 		</Grid.RowDefinitions>
 		<StackPanel Spacing="8" BorderBrush="Black" BorderThickness="2" Padding="4">
-			<Slider Header="Deptness" x:Name="deptness" Minimum="10" Maximum="150" Value="50" />
+			<Slider Header="Deepness" x:Name="deepness" Minimum="10" Maximum="150" Value="50" />
```
Comment (jeromelaban) [inline src/SamplesApp/UITests.Shared/Windows_UI_Xaml/UIElementTests/UIElement_MeasurePerf.xaml:16]: Depth

File: src/Uno.UI/Extensions/ViewExtensions.cs [hunk 8]
```diff
@@ -50,23 +51,23 @@ internal static string GetElementSpecificDetails(this UIElement element)
...
+				string? rows = default;
 				if (grid.RowDefinitions.Count > 1)
 				{
 					rows = " Rows=" + grid.RowDefinitions
 						.Select<RowDefinition, string>(x => x.Height.ToString())
 						.JoinBy(",");
 				}
 
-				return columns + rows;
+				return "" + columns + rows;
```
Comment (dr1rrb) [inline src/Uno.UI/Extensions/ViewExtensions.cs:70]: ???

File: src/Uno.UI/UI/Xaml/Controls/Grid/Grid.h.cs [hunk 9]
```diff
@@ -103,16 +103,31 @@ internal SpanStoreStackVector() : base(c_spanStoreStackVectorSize)
...
+		bool TrySetGridFlags(GridFlags mask)
+		{
+			if (HasGridFlags(mask))
+			{
+				return false;
+			}
+
+			m_gridFlags |= mask;
+			return true;
+		}
```
Comment (dr1rrb) [inline src/Uno.UI/UI/Xaml/Controls/Grid/Grid.h.cs:122]: I would move this out of the .h.cs

File: src/Uno.UI/UI/Xaml/Controls/ScrollContentPresenter/NativeScrollContentPresenter.Android.cs [hunk 10]
```diff
@@ -103,32 +103,49 @@ private void InitializeScrollbars()
 			}
 		}
 
-		//TODO generated code
+		ILayouter ILayouterElement.Layouter => _layouter;
+		Size ILayouterElement.LastAvailableSize => LayoutInformation.GetAvailableSize(this);
+		bool ILayouterElement.IsMeasureDirty => true;
+		bool ILayouterElement.IsFirstMeasureDone => false;
```
Comment (dr1rrb) [inline src/Uno.UI/UI/Xaml/Controls/ScrollContentPresenter/NativeScrollContentPresenter.Android.cs:109]: Sounds weird this is always `false` (idem for `IsMeasureDirty`)

File: src/Uno.UI/UI/Xaml/IFrameworkElement.cs [hunk 11]
```diff
@@ -191,29 +192,48 @@ public static DependencyObject GetTemplateChild(this IFrameworkElement e, string
...
+					break;
+				case View view:
 #if XAMARIN_ANDROID
-			var UnoViewGroup = e as UnoViewGroup;
+					view.RequestLayout();
+
+					// Invalidate the first "managed" parent to
+					// ensure its .MeasureOverride() gets called
+					var parent = view.Parent;
+					while (parent is { })
```
Comment (dr1rrb) [inline src/Uno.UI/UI/Xaml/IFrameworkElement.cs:213]:
```suggestion
					while (parent is not null)
```

**Review by dr1rrb — APPROVED** [review]

**Pull request merged**